caikit/interfaces/nlp/tasks.py

```diff
@@ -32,3 +40,41 @@ class TextGenerationTask(TaskBase):
     """The Text Generation Task is responsible for taking input prompting text
     and generating additional text from that prompt.
     """
+
+
+@task(
+    required_parameters={"text": str},
+    output_type=ClassificationResult,
+)
+class TextClassificationTask(TaskBase):
+    pass
+
+
+@task(
+    unary_parameters={"text": str},
+    streaming_parameters={"text_stream": Iterable[str]},
+    unary_output_type=TokenClassificationResult,
+    streaming_output_type=Iterable[StreamingTokenClassificationResult],
+)
+class TokenClassificationTask(TaskBase):
+    pass
+
+
+@task(
+    required_parameters={"text": str},
+    output_type=TokenizationResult,
+)
+class TokenizationTask(TaskBase):
+    pass
+
+
+@task(
+    unary_parameters={"text": str},
+    unary_output_type=ClassifiedGeneratedTextResult,
+    streaming_output_type=Iterable[ClassifiedGeneratedTextStreamResult],
+)
+class ClassificationWithTextGenerationTask(TaskBase):
```

Comment:
Is this what we're using for guardrail classifiers? @evaline-ju I'm curious where our thinking is around having `guarded-text-generation` as a `task` versus keeping it generic like it is here. The data models for something like "generated text with references to entities mentioned in the text" would match this same signature, but as a grouping of models, I think this would be serving a different purpose, so I would imagine that it might make sense to define the `task` based on the semantics and the `dataobject`s based on the data shape. Thoughts?

Reply:
I think the modules here are really only "text classification", the model's and module's functionality isn't specific to anything guardrail related and is more related to token classification or sentence classification. Having a generic name here would allow us to in future use the same structure / modules to run other task classification tasks on generation.

Other than that, I think the term "gaurded" may become a misnomer, since we do not have currently functionalities around that and also the depth of "gaurded" is yet to be discovered